web/src/typing_events.js

```diff
@@ -77,6 +79,22 @@ export function render_notifications_for_narrow() {
     }
 }
 
+function render_message_editing_typing(message_id, is_typing) {
+    const $row = message_lists.current.get_row(message_id);
+    const $editing_notifications = $row.find("#editing_notifications");
+    if (is_typing) {
+        $row.find(".message_edit_notice").hide();
+    } else {
+        $row.find("#typing-animation").hide();
+        $row.find(".message_edit_notice").show();
+    }
+    $editing_notifications.html(
+        render_editing_notifications({
+            typing: is_typing,
+        }),
+    );
+}
```

Comment:
Does this logic correctly handle multiple rendered message lists? You can likely test using the `s` keyboard shortcut to flip between "All messages" (which is always rendered) and a narrowed view.

Reply:
I tested it manually.... it does handle the "All messages" list and narrowed ones (to both stream and topic) correctly.

Reply:
yup, you're right. This logic works as far as the container is re rendered when you change narrows/views. However, for "All Messages", since it is already present and we don't re render anything.... I'll try to think of something else.